tests/integration/test_kfp_generic_installation.py

```diff
@@ -0,0 +1,350 @@
+    """Create a run and monitor it to completion."""
+    # Create a run, save response in variable for easy manipulation
+    run_response = create_run_from_pipeline_file
+
+    # FIXME: wait for completion does not work at the moment, it seems like
+    # the status of the run is never updated and is causing this to timeout
+    # Monitor the run to completion, the pipeline should not be executed in
+    # more than 60 seconds as it is a very simple operation
+    # monitor_response = kfp_client.wait_for_run_completion(run_response.run_id, timeout=60)
+
+    # assert monitor_response.success is True
+
+
+# ---- ScheduledWorfklows and Argo focused test case
+async def test_create_and_monitor_recurring_run(create_recurring_run, kfp_client):
+    """Create a recurring run and monitor it to completion."""
+    # Create a recurring run, save response in variable for easy manipulation
+    recurring_job = create_recurring_run
+
+    # Assert the job is enabled
+    assert recurring_job.enabled is True
+
+    # Assert the job executes once every two seconds
+    assert recurring_job.trigger.cron_schedule.cron == "*/2 * * * * *"
+
+    # Wait for the recurring job to schedule some runs
+    # time.sleep(6)
+
+    # FIXME: disabling the job does not work at the moment, it seems like
+    # the status of the recurring run is never updated and is causing the
+    # following assertion to fail
+    # Disable the job after few runs
+    # kfp_client.disable_job(recurring_job.id)
+
+    # Assert the job is disabled
+    # assert recurring_job.enabled is False
+
+
+# ---- KFP Viewer focused test cases
+async def test_apply_sample_viewer(lightkube_client, apply_viewer):
+    """Test a Viewer can be applied and its presence is verified."""
+    viewer_object = apply_viewer
+
+    viewer = lightkube_client.get(
+        res=type(viewer_object),
+        name=viewer_object.metadata.name,
+        namespace=viewer_object.metadata.namespace,
+    )
+    assert viewer is not None
+
+
+# ---- KFP Visualization focused test cases
+async def test_viz_server_healthcheck(ops_test: OpsTest):
+    """Run a healthcheck on the server endpoint."""
+    status = await ops_test.model.get_status()
+    units = status["applications"]["kfp-viz"]["units"]
+    url = units["kfp-viz/0"]["address"]
+    headers = {"kubeflow-userid": "user"}
+    result_status, result_text = await fetch_response(url=f"http://{url}:8888", headers=headers)
+
+    assert result_status == 200
+
+
+@pytest.mark.skip("The Authentication workflow prevents this test from running")
+async def test_create_viz(ops_test: OpsTest, auth_session):
+    """Create a Visualization."""
+    # Get session cookies
+    session_cookies = auth_session["session_cookie"]
+
+    # Get Visualization server information
+    status = await ops_test.model.get_status()
+    units = status["applications"]["kfp-viz"]["units"]
+    url = units["kfp-viz/0"]["address"]
```

Comment:
this might make for a nice helper like:
```
async def get_address_from_unit(model, unit_name: str):
  app_name = "".join(unit_name.split("/")[:-1])
  status = await model.get_status()
  units = status["applications"][app_name]["units"]
  return units[unit_name]["address"]

def test_something(ops_test):
  ...
  url = await get_address_from_unit(ops_test.model, "kfp-viz/0")
```

Reply:
Agreed, will update.